From 929f99966f2f56664b386829347c2a89b06deed9 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Thu, 12 Oct 2023 20:20:48 +0000
Subject: [PATCH] libstdc++: use std::nullptr_t in FragmentDataIterator

nullptr_t without namespace std is not defined in libstdc++.

Bug: 957519
Change-Id: I831e4c8e37e6c3790141f8dabe56a3066f01d79c
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4936372
Auto-Submit: Stephan Hartmann <stha09@googlemail.com>
Reviewed-by: Florin Malita <fmalita@chromium.org>
Commit-Queue: Florin Malita <fmalita@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1209035}
---
 .../blink/renderer/core/paint/fragment_data_iterator.h        | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/third_party/blink/renderer/core/paint/fragment_data_iterator.h b/third_party/blink/renderer/core/paint/fragment_data_iterator.h
index c42fba889866d98..56a7a45d72d6a66 100644
--- a/third_party/blink/renderer/core/paint/fragment_data_iterator.h
+++ b/third_party/blink/renderer/core/paint/fragment_data_iterator.h
@@ -52,7 +52,7 @@ class FragmentDataIterator
  public:
   explicit FragmentDataIterator(const LayoutObject& object)
       : FragmentDataIteratorBase(&object.FirstFragment()) {}
-  explicit FragmentDataIterator(nullptr_t)
+  explicit FragmentDataIterator(std::nullptr_t)
       : FragmentDataIteratorBase(nullptr) {}
 };
 
@@ -63,7 +63,7 @@ class MutableFragmentDataIterator
   explicit MutableFragmentDataIterator(const LayoutObject& object)
       : FragmentDataIteratorBase(
             &object.GetMutableForPainting().FirstFragment()) {}
-  explicit MutableFragmentDataIterator(nullptr_t)
+  explicit MutableFragmentDataIterator(std::nullptr_t)
       : FragmentDataIteratorBase(nullptr) {}
 };
 
